app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,355 @@
+import androidx.core.graphics.BlendModeColorFilterCompat;
+import androidx.core.graphics.BlendModeCompat;
+import androidx.core.splashscreen.SplashScreen;
+import androidx.recyclerview.widget.RecyclerView;
+import protect.card_locker.preferences.SettingsActivity;
+
+public class ManageGroupActivity extends CatimaAppCompatActivity implements ManageGroupCursorAdapter.CardAdapterListener
+{
+    private static final String TAG = "Catima";
+
+    private final DBHelper mDB = new DBHelper(this);
+    private ManageGroupCursorAdapter mAdapter;
+    private ActionMode mCurrentActionMode;
+    private Menu mMenu;
+
+    // currently unused
+    protected String mFilter = "";
+    protected DBHelper.LoyaltyCardOrderDirection mOrderDirection = DBHelper.LoyaltyCardOrderDirection.Ascending;
+    protected DBHelper.LoyaltyCardOrder mOrder = DBHelper.LoyaltyCardOrder.Alpha;
+
+    protected Group mGroup = null;
+    private RecyclerView mCardList;
+    private View mHelpText;
+    private View mNoMatchingCardsText;
+    private View mNoGroupCardsText;
+    private EditText mGroupNameText;
+    private TextView mGroupNameLabel;
+    private ActionBar mActionBar;
+
+    private HashMap<Integer, Boolean> mAdapterState;
+    private String mCurrentGroupName;
+
+    private boolean mGroupNameNotInUse;
+
+    private boolean mDarkMode;
+
+    @Override
+    protected void onCreate(Bundle inputSavedInstanceState)
+    {
+        super.onCreate(inputSavedInstanceState);
+        setContentView(R.layout.activity_manage_group);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        mHelpText = findViewById(R.id.helpText);
+        mNoMatchingCardsText = findViewById(R.id.noMatchingCardsText);
+        mNoGroupCardsText = findViewById(R.id.noGroupCardsText);
+        mCardList = findViewById(R.id.list);
+
+        mGroupNameText = findViewById(R.id.editTextGroupName);
+        mGroupNameLabel = findViewById(R.id.textViewEditGroupName);
+
+        mAdapter = new ManageGroupCursorAdapter(this, null, this);
+        mCardList.setAdapter(mAdapter);
+        registerForContextMenu(mCardList);
+
+        mGroup = null;
+
+        mDarkMode = Utils.isDarkModeEnabled(getApplicationContext());
+
+        if (inputSavedInstanceState != null) {
+            ManageGroupActivityInGroupState adapterState = inputSavedInstanceState.getParcelable("mAdapterState");
+            if (adapterState != null) {
+                mAdapterState = adapterState.getMap();
+            }
+            mCurrentGroupName = inputSavedInstanceState.getString("mCurrentGroupName");
+        }
+
+        mActionBar = getSupportActionBar();
+        if (mActionBar == null){
+            throw(new RuntimeException("mActionBar is not expected to be null here"));
+        }
+        mActionBar.setDisplayHomeAsUpEnabled(true);
+        mActionBar.setDisplayShowHomeEnabled(true);
+    }
+
+    private void resetGroupNameTextColor() {
```

Comment:
This looks like a hack around incorrect XML theming.

Reply:
This seems to be the way for AppCompatActivity's toolbar instance, if the focus here is line 119
If you wish to and have suggestions on how to do that in another way please let me know

Reply:
as for line 122 that was for resetting the text color that turns red when a duplicated group name is entered
is using colors from R.colors good enough?

Reply:
Hmm... I don't know anymore what I was exactly responding to here :(

Anyway, instead of manually editing colours, you probably want to use `setError` as provided by TextInputLayout: https://developer.android.com/reference/com/google/android/material/textfield/TextInputLayout#setError(java.lang.CharSequence)